codegen type issue

`aztec codegen` (as of at least 0.57.0) is generating code that typescript complains about.

Right now a number of us are working around this with:
`sed -i -e 's/as NoirCompiledContract/as unknown as NoirCompiledContract/g' ${CONTRACT}.ts`.

Discord discussion: https://discord.com/channels/1144692727120937080/1298434944456069151


```
src/contracts/Token.ts:38:59 - error TS2352: Conversion of type '{ transpiled: boolean; noir_version: string; name: string; functions: ({ name: string; is_unconstrained: boolean; custom_attributes: string[]; abi: { error_types: { "16137133351287663689"?: undefined; "14336010898202591232"?: undefined; "3469988399892582431"?: undefined; }; parameters: ({ ...; } | { ...; })[]; retur...' to type 'NoirCompiledContract' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Types of property 'functions' are incompatible.
    Type '({ name: string; is_unconstrained: boolean; custom_attributes: string[]; abi: { error_types: { "16137133351287663689"?: undefined; "14336010898202591232"?: undefined; "3469988399892582431"?: undefined; }; parameters: ({ ...; } | { ...; })[]; return_type: null; }; bytecode: string; debug_symbols: string; brillig_name...' is not comparable to type 'NoirFunctionEntry[]'.
      Type '{ name: string; is_unconstrained: boolean; custom_attributes: string[]; abi: { error_types: { "16137133351287663689"?: undefined; "14336010898202591232"?: undefined; "3469988399892582431"?: undefined; }; parameters: ({ ...; } | { ...; })[]; return_type: null; }; bytecode: string; debug_symbols: string; brillig_names...' is not comparable to type 'NoirFunctionEntry'.
        Type '{ name: string; is_unconstrained: boolean; custom_attributes: never[]; abi: { error_types: { "3469988399892582431": { error_kind: string; item_types: never[]; length: number; }; "16137133351287663689"?: undefined; "14336010898202591232"?: undefined; }; parameters: ({ ...; } | ... 1 more ... | { ...; })[]; return_typ...' is not comparable to type 'NoirFunctionEntry'.
          The types of 'abi.error_types' are incompatible between these types.
            Type '{ "3469988399892582431": { error_kind: string; item_types: never[]; length: number; }; "16137133351287663689"?: undefined; "14336010898202591232"?: undefined; }' is not comparable to type 'Record<string, AbiErrorType>'.
              Property '"16137133351287663689"' is incompatible with index signature.
                Type 'undefined' is not comparable to type 'AbiErrorType'.

38 export const TokenContractArtifact = loadContractArtifact(TokenContractArtifactJson as NoirCompiledContract);                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/noir/noir-repo/tooling/noir_js_types/src/types.ts
+++ b/noir/noir-repo/tooling/noir_js_types/src/types.ts
@@ -39,7 +39,7 @@ export type WitnessMap = Map<number, string>;
 export type Abi = {
   parameters: AbiParameter[];
   return_type: { abi_type: AbiType; visibility: Visibility } | null;
-  error_types: Record<string, AbiErrorType>;
+  error_types: Partial<Record<string, AbiErrorType>>;
 };
 
 export interface VerifierBackend {

--- a/yarn-project/types/src/noir/index.ts
+++ b/yarn-project/types/src/noir/index.ts
@@ -40,7 +40,7 @@ export interface NoirFunctionAbi {
     visibility: ABIParameterVisibility;
   };
   /** Mapping of error selector => error type */
-  error_types: Record<string, AbiErrorType>;
+  error_types: Partial<Record<string, AbiErrorType>>;
 }
 
 /**

